src/rngs/mod.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// https://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Various RNGs and wrappers.
+//!
+//! This module provides the three standard RNGs exposed by Rand (besides
+//! specific PRNG algorithms in the `prng` module). Those are [`ThreadRng`],
+//! [`StdRng`], and [`SmallRng`].
+//!
+//! To get a seed for those PRNGs at runtime, [`EntropyRng`], [`OsRng`] and
+//! [`JitterRng`] are sources of external randomness. Those are mostly
+//! implementation details for Rand, prefer to use the [`FromEntropy`] trait
+//! instead.
+//!
+//! [`ReseedingRng`] is a wrapper around another PRNG that reseeds it after it
+//! has generated a certain number of random bytes. Using it be an advantage in
+//! specific circumstances.
+//!
+//! [`ReadRng`] is an adapter to turn any `Read` into an RNG.
+//!
+//! Finally there is [`StepRng`], which is not much more than a counter
```

Comment:
Can we call this `mock::StepRng` please? It emphasises that this is for testing only, and we might add more mock RNGs later. I.e. don't re-export `StepRng` below.